pkg/driver/controller_modify_volume.go

```diff
@@ -107,55 +113,106 @@ func mergeModifyVolumeRequest(input modifyVolumeRequest, existing modifyVolumeRe
 		}
 		existing.modifyDiskOptions.VolumeType = input.modifyDiskOptions.VolumeType
 	}
-
+	if input.modifyTagsOptions != nil {
+		if len(input.modifyTagsOptions.TagsToAdd) > 0 || len(input.modifyTagsOptions.TagsToDelete) > 0 {
+			if existing.modifyTagsOptions != nil && input.modifyTagsOptions != existing.modifyTagsOptions {
+				return existing, fmt.Errorf("Different tags were requested by a previous request. Current: %v, Requested: %v", existing.modifyTagsOptions, input.modifyTagsOptions)
+			}
+			existing.modifyTagsOptions = &cloud.ModifyTagsOptions{
+				TagsToAdd:    input.modifyTagsOptions.TagsToAdd,
+				TagsToDelete: input.modifyTagsOptions.TagsToDelete,
+			}
+		}
+	}
 	return existing, nil
 }
 
+func executeModifyTagsRequest(volumeID string, options modifyVolumeRequest, c cloud.Cloud) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+	if options.modifyTagsOptions != nil {
+		err := c.ModifyTags(ctx, volumeID, options.modifyTagsOptions.TagsToAdd, options.modifyTagsOptions.TagsToDelete)
+		if err != nil {
+			if errors.Is(err, cloud.ErrInvalidArgument) {
+				return status.Errorf(codes.InvalidArgument, "Could not modify volume tags (invalid argument) %q: %v", volumeID, err)
+			}
+			return status.Errorf(codes.Internal, "Could not modify volume tags %q: %v", volumeID, err)
+		}
+	} else {
+		klog.InfoS("No tags to modify", "volumeID", volumeID)
```

Comment:
Would increasing the V level at which this informational is logged be more appropriate?

Reply:
I removed it completely as it was a log message that I was using when I was doing testing, there is no log message for any of the other modifications when there are none present(ex. "No IOPS to modify") so it would be inconsistent to leave it as well.